src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameGapCheck.java

```diff
@@ -0,0 +1,218 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.direction.EdgeDirectionComparator;
+
+/**
+ * This check identifies Edges with no name Tag that are between two other Edges with the same name
+ * Tag, OR the Edge has a name Tag but does not equal the name Tag of the Edges that it is between.
+ *
+ * @author sugandhimaheshwaram
+ */
+public class RoadNameGapCheck extends BaseCheck
+{
+    /**
+     * This checks if the connected edges have same direction or not.
+     */
+    class EdgePredicate
+    {
+        private Edge edge;
+        private EdgeDirectionComparator edgeDirectionComparator = new EdgeDirectionComparator();
+
+        EdgePredicate(final Edge edge)
+        {
+            this.edge = edge;
+        }
+
+        boolean isSameHeading(final AtlasObject object)
+        {
+            final Edge otherEdge = (Edge) object;
+            return object != null
+                    && this.edgeDirectionComparator.isSameDirection(this.edge, otherEdge, false);
+        }
+    }
+
+    // You can use serialver to regenerate the serial UID.
+    private static final long serialVersionUID = 7104778218412127847L;
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList("Edge name is empty.",
+            "Edge name {1} is different from in edge name and out edge name.");
+
+    /**
+     * Tests if the {@link AtlasObject} has a highway tag that do contain TERTIARY, SECONDARY,
+     * PRIMARY, TRUNK, or MOTORWAY
+     */
+    private static final Predicate<AtlasObject> VALID_HIGHWAY_TAG = object -> Validators
+            .hasValuesFor(object, HighwayTag.class)
```

Comment:
I think this is redundant when using Validators.isOfType().

Reply:
Okay. I will take this out.